frame/support/test/tests/construct_runtime.rs

```diff
@@ -334,12 +334,23 @@ mod origin_test {
 
 		origin.set_caller_from(Origin::root());
 		assert!(matches!(origin.caller, OriginCaller::system(super::system::RawOrigin::Root)));
+
+		// Root origin bypass all filter.
+		assert_eq!(origin.filter_call(&accepted_call), true);
+		assert_eq!(origin.filter_call(&rejected_call), true);
+
+		origin.set_caller_from(Origin::from(Some(0)));
+
+		// Back to another signed origin, the filtered are now effective again
 		assert_eq!(origin.filter_call(&accepted_call), false);
```

Comment:
```suggestion
		assert_eq!(origin.filter_call(&accepted_call), true);
```

Shouldn't the accepted call always be true?

Reply:
just above we set the caller to a signed origin, the filter are effective again

Reply:
ah I understand what you mean, ok I'll rewrite the end of the test to make it more clear

Reply:
refactored in https://github.com/paritytech/substrate/pull/9948/commits/1411f7b030e4e20c63ff4881afe1fe3c11b2bc2f